website/docs/pipelines/promoting-applications.mdx

```diff
@@ -130,133 +133,191 @@ Be sure to create the Provider/Alert tuple on **each of the leaf clusters
+3. Prefer granting access to specific secrets, rather than granting `list` and `watch` on secrets as:
 
-### Credentials Secret
+> It is also important to note that list and watch access also effectively allow users to read Secret contents.
 
-In the journey of creating a pull request, there are different secrets involved:
+#### Policy
 
-1. Pipeline controller receives events via [webhook from leaf clusters](./#setup-notifications-from-leaf-clusters). Hmac is used for authN/authZ so an hmac key should be provided in this case.
-2. Pipeline controller clones and patches manifests to promote from the pipeline configuration repo. A set of [git credentials](https://fluxcd.io/flux/components/source/gitrepositories/#secret-reference) are required.
-3. Pipeline controller uses git provider api to create the pull request with the promoted manifests. A Personal Access Token (PAT) needs to be created to interact with pipelines git provider API. This PAT is also used to list pull requests from the configured repository.
+By following the guidelines above, you can have a safe initial configuration. However, given there are no deny semantics in [RBAC](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole), you need to guard future changes.
 
-Create a Kubernetes secret with the previous data.
+> An RBAC Role or ClusterRole contains rules that represent a set of permissions.
+> Permissions are purely additive (there are no "deny" rules).
 
-<details><summary>Expand to see example</summary>
+You should ensure that attempts to break this contract are blocked and detected. You could achieve it
+by using Weave GitOps' [Policy capabilities](../../policy/intro/). The Policy Agent acts in two complementary modes:
+- [Admission Controller](../../policy/intro#admission-controller) protects from any attempt to create non-compliant RBAC resources that would end granting access to the secret.
+- [Audit](../../policy/intro#audit) helps you identify already existing resources that are out of compliance. For example,
+roles created before policy agent was introduced as admission controller.
 
-```shell
-# example to use git over https with basic auth and pat
-$ kubectl create secret generic promotion-credentials \
-  --namespace=pipeline-01 \
-  --from-literal="username=<bot account name>" \
-  --from-literal="password=<token value>" \
-  --from-literal="token=<token value>" \
-  --from-literal="hmac-key=<hmac-key value>"
-```
+Once you have enabled Policy, the [Policy Library](../../policy/weave-policy-profile/) gives you a set of
+good practices policies that will help you keep pipeline secrets secure according to the previous RBAC recommendations. Deploy
+them as Kustomization based on the following example:
+
+:::tip
+In case you don't have access to the Policy Library, work with your Weaveworks Technical Account Manager (TAM) or
+Weaveworks Customer Reliability Engineer (CRE) to help with this step.
```

Comment:
@squaremo let me know your thoughts

<img width="854" alt="Screenshot 2023-07-26 at 11 46 02" src="https://github.com/weaveworks/weave-gitops/assets/12957664/f109397d-4dfa-48e8-940a-2fba6e757e38">

Reply:
If those are real job titles, I'm on board with it :-)